plugins/linux-capture/pipewire.c

```diff
@@ -116,6 +128,41 @@ struct dbus_call_data {
 
 /* auxiliary methods */
 
+static bool parse_pw_version(struct obs_pw_version *dst, const char *version)
+{
+	int n_matches = sscanf(version, "%d.%d.%d", &dst->major, &dst->minor,
+			       &dst->micro);
+	return n_matches == 3;
+}
+
+static bool check_pw_version(const struct obs_pw_version *pw_version, int major,
+			     int minor, int micro)
+{
+	return pw_version->major >= major && pw_version->minor >= minor &&
+	       pw_version->micro >= micro;
+}
+
+static bool update_pw_versions(obs_pipewire_data *obs_pw, const char *version)
+{
+	bool ret = parse_pw_version(&obs_pw->library_version,
+				    pw_get_library_version());
+	ret &= parse_pw_version(&obs_pw->header_version,
+				pw_get_headers_version());
+	ret &= parse_pw_version(&obs_pw->server_version, version);
+
+	blog(LOG_INFO, "[pipewire] server_version %d.%d.%d",
+	     obs_pw->server_version.major, obs_pw->server_version.minor,
+	     obs_pw->server_version.micro);
+	blog(LOG_INFO, "[pipewire] library_version %d.%d.%d",
+	     obs_pw->library_version.major, obs_pw->library_version.minor,
+	     obs_pw->library_version.micro);
+	blog(LOG_INFO, "[pipewire] header_version %d.%d.%d",
+	     obs_pw->header_version.major, obs_pw->header_version.minor,
+	     obs_pw->header_version.micro);
+
+	return ret;
+}
+
```

Comment:
Allow me to suggest writing this function like this:

```c
static void update_pw_versions(obs_pipewire_data *obs_pw, const char *server_version)
{
	blog(LOG_INFO, "[pipewire] server version: %s", server_version);

	if (!parse_pw_version(&obs_pw->server_version, server_version))
		blog(LOG_WARNING, "[pipewire] couldn't parse server version");
}
```

(Note: I didn't run clang-format on this)